happi/audit.py

```diff
@@ -0,0 +1,424 @@
+from happi.containers import registry
+from enum import Enum
+
+logger = logging.getLogger(__name__)
+
+
+class ReportCode(Enum):
+    """
+    Class to define report codes
+    """
+    SUCCESS = 0
+    INVALID = 1
+    MISSING = 2
+    EXTRAS = 3
+    NO_CODE = 9
+
+
+class Command(object, metaclass=ABCMeta):
+    """
+    Bluprint for the command class
+    This will be useful if all the commands are inheriting from it
+    """
+
+    def __init__(self, name, summary):
+        self.name = name
+        self.summary = summary
+
+    @abstractmethod
+    def add_args(self, parser):
+        """
+        Adds arguments to a parser
+
+        Parameters
+        -----------
+        parser: ArgumentParser
+            Parser to add arguments to
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def run(self, args):
+        """
+        Parses the arguments given to the command.
+        And hanles the logic for this command
+
+        Parameters
+        -----------
+            args: Namespace
+        """
+        raise NotImplementedError
+
+
+class Audit(Command):
+    """
+    Audits the database for valid entries.
+    """
+    def __init__(self):
+        self.name = 'audit'
+        self.help = "Inspect a database's entries"
+        self._all_devices = set()
+        self._all_items = []
+        # 9 - NO_CODE
+        self.report_code = ReportCode(9)
+
+    def add_args(self, cmd_parser):
+        cmd_parser.add_argument(
+            "--file", help='Path to the json file (database) to be audited'
+        )
+        cmd_parser.add_argument(
+            "--extras", action='store_true', default=False,
+            help='Check specifically for extra attributes'
+        )
+
+    def run(self, args):
+        """
+        Validate the database passed in with --file option
+        """
+        if args.file is not None:
+            self.process_args(args.file, args)
+        else:
+            """
+            Validate the database defined in happi.cfg file
+            """
+            path_to_config = Client.find_config()
+
+            if path_to_config:
+                logger.info('Using Client cfg path %s ', path_to_config)
+                config = ConfigParser()
+                config.read(path_to_config)
+                try:
+                    database_path = config.get('DEFAULT', 'path')
+                except Exception as e:
+                    logger.error('Error when trying '
+                                 'to get database path %s', e)
+                    return
+                else:
+                    self.process_args(database_path, args)
+            else:
+                logger.error('Could not find the Happi Configuration file')
+                sys.exit(1)
+
+    def process_args(self, database_path, args):
+        """
+        Checks to see if a valid path to database was provided.
+        If --extras is provided, will call for check xtra attributes, otherwise
```

Comment:
```suggestion
        If --extras is provided, will check extra attributes, otherwise
```

Additionally, `--extras` is a CLI argument, not a Python [keyword] argument. These should be functions with Python arguments, as I mentioned in the comment above.